File: experiments/009-CuisByScripts/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260916-095426-6223/analysis/own-tests-project/src/main/java/surveycrawler/CrawlerFacingLeft.java
package surveycrawler;

public class CrawlerFacingLeft extends CrawlerFacing {

    // facing name

    @Override
    public String facingName() {
        return "Left";
    }

    // moving

    @Override
    public void retreat(SurveyCrawler aCrawler) {
        aCrawler.moveRight();
    }

    @Override
    public void advance(SurveyCrawler aCrawler) {
        aCrawler.moveLeft();
    }

    // facing

    @Override
    public void turnCounterClockwise(SurveyCrawler aCrawler) {
        aCrawler.faceDown();
    }

    @Override
    public void turnClockwise(SurveyCrawler aCrawler) {
        aCrawler.faceUp();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-213927-34997/exercise/python/src/surveycrawler/crawler_facing_left.py
from surveycrawler.crawler_facing import CrawlerFacing


class CrawlerFacingLeft(CrawlerFacing):

    # facing name

    @classmethod
    def facing_name(cls):
        return "Left"

    # moving

    def retreat(self, a_crawler):
        a_crawler.move_right()

    def advance(self, a_crawler):
        a_crawler.move_left()

    # facing

    def turn_counter_clockwise(self, a_crawler):
        a_crawler.face_down()

    def turn_clockwise(self, a_crawler):
        a_crawler.face_up()
